### PR TITLE
PLU-66 (feat): tag S3 attachments with AttachmentType for lifecycle management

### PR DESCRIPTION
**Pull request by kevinkim-ogp**
## TL;DR

Tags all S3 attachment uploads with an `AttachmentType` object tag so that S3 lifecycle rules can expire transitive attachments (FormSG submissions, LetterSG PDFs) automatically. This is the backend half of a two-part change — the lifecycle rules themselves live in `plumber-infra`.

## What changed?

- **`packages/backend/src/helpers/s3.ts`**
  - Added `ATTACHMENT_TYPE` constant (`transitive | test | manual`)
  - Added optional `tags` parameter to `putObject()` — encoded as a URL query string and passed to `PutObjectCommand` as `Tagging`
  - `getPresignedPost()` now always includes `AttachmentType=manual` in the presigned POST XML tagging field (previously only the dev GuardDuty mock tag was conditionally included; the two tags are now merged)

- **`packages/backend/src/apps/formsg/auth/helpers/store-attachment-in-s3.ts`**
  - Tags based on `$.flow.isActive` (not `$.execution.testRun`). FormSG attachment uploads happen inside `verifyWebhook`, which runs before `processTrigger` creates the execution — so `$.execution.testRun` is always `false` at that point. `$.flow.isActive` is the correct signal: active = `transitive`, inactive = `test`.

- **`packages/backend/src/apps/lettersg/helpers/attachment.ts`**
  - Tags based on `$.execution.testRun` — correct here because LetterSG runs during action execution in a worker where the execution is fully established

## Tests

- [ ] Trigger a FormSG flow with a file attachment on a **published** pipe — confirm the S3 object has tag `AttachmentType=transitive`
- [ ] Submit a real FormSG form on an **unpublished** pipe — confirm the tag is `AttachmentType=test`
- [ ] Run a **test execution** (via the test button) of a FormSG flow with a file attachment — no new S3 upload happens (replays from last submission); confirm no errors
- [ ] Trigger a LetterSG `create-letter` action (with `shouldGeneratePdf=true`) in a published pipe — confirm the PDF object is tagged `AttachmentType=transitive`
- [ ] Upload a manual attachment via the Plumber UI — confirm the object has `AttachmentType=manual`
- [ ] Confirm the GuardDuty scan tag (`GuardDutyMalwareScanStatus`) is still present on manual uploads in dev

## Deployment notes

> ⚠️ **This PR must be deployed _after_ the corresponding `plumber-infra` change is merged and applied.**

The `plumber-infra` change adds `s3:PutObjectTagging` to the ECS task role IAM policy. Without it, every `putObject` call with tags and every manual upload presigned POST will fail with `AccessDenied`.

Deploy order:
1. Merge and apply the `plumber-infra` IAM policy change → [opengovsg/plumber-infra#105](https://github.com/opengovsg/plumber-infra/pull/105)
2. Deploy this PR

### CODE REVIEW

**Comment by kevinkim-ogp** [comment]
* **#1710** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1710?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1710?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop-v2`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>lfg</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/opengovsg?ref=merge-queue-instructions-in-stack-comment&prId=1710).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend/src/apps/formsg/auth/helpers/store-attachment-in-s3.ts [hunk 1]
```diff
@@ -39,9 +44,12 @@ async function storeAttachmentInS3(
     metadata: {
       flowId: $.flow.id,
       stepId: $.step.id,
-      executionId: $.execution.id ?? '', // Empty = test runs.
+      executionId: $.execution.id ?? '',
       formId: $.auth.data.formId as string,
     },
+    tagging: buildAttachmentTypeTagging(
+      $.flow.active ? ATTACHMENT_TYPE.TRANSITIVE : ATTACHMENT_TYPE.TEST,
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/formsg/auth/helpers/store-attachment-in-s3.ts]: at this point, we don't have `$.execution.testRun`? Edge case: user might publish flow before this code runs.

Very unlikely but just saying `$.execution` seems to be most accurate

**Review by ogp-weeloong — APPROVED** [review]
overall ok, have qn on the `$.execution`

File: packages/backend/src/apps/formsg/auth/helpers/store-attachment-in-s3.ts (hunk 1 above)
Comment (kevinkim-ogp) [reply packages/backend/src/apps/formsg/auth/helpers/store-attachment-in-s3.ts]: issue with formsg submission is that `$.execution.testRun = false` when its a real form submission and the form is unpublished
